rpc_proxy.go

```diff
@@ -135,6 +143,41 @@ func (p *rpcProxy) Start() error {
 		return fmt.Errorf("could not dial lnd: %v", err)
 	}
 
+	// Make sure we can connect to all the daemons that are configured to be
+	// running in remote mode.
+	if p.cfg.FaradayMode == ModeRemote {
+		p.faradayConn, err = dialBackend(
+			"faraday", p.cfg.Remote.Faraday.RPCServer,
+			lncfg.CleanAndExpandPath(
+				p.cfg.Remote.Faraday.TLSCertPath,
+			),
+		)
+		if err != nil {
+			return fmt.Errorf("could not dial remote faraday: %v",
+				err)
+		}
+	}
+
+	if p.cfg.LoopMode == ModeRemote {
```

Comment:
I was trying to make it as similar as possible to `lnd-mode`. But because that flag defaults to the value `"remote"` and has other implications as well (how/where the config/logs/certs are stored), it's probably okay to use a different pattern for the other daemons. Switched them over to a `bool`.

Reply:
I mean, why not use bool for `lnd-mode` also? (rename it to `remote-lnd` or similar.

Reply:
The initial idea was to make it more verbose for the user that just looks at the CLI help output. With an explicit mode you can have better descriptions in the section headers.
For example:
```go
Remote *RemoteConfig `group:"Remote mode options (use when lnd-mode=remote)" namespace:"remote"`
```

Booleans are also tricky because they need to be used differently depending on whether you're in a config file or not (`bool-flag=false` vs. not specifying the flag on the CLI). Additionally, the default for `lnd` is remote. So if we added a `--lnd-integrated` boolean flag, the above help text would become a bit less easy to understand IMO:
```go
Remote *RemoteConfig `group:"Remote mode options (use when lnd-integrated=false or not present as a flag)" namespace:"remote"`
```

And finally, I didn't want to break existing users that currently use the `--lnd-mode=integrated` flag.

Reply:
ah, I see. I would say use the string variant for all then, nice to have them be similar :)